### PR TITLE
Replaced linting tools with with ruff and added ruff formatter

### PR DESCRIPTION
**Pull request by santigandolfo**
Currently, the repo uses isort, flake8, flake8-bugbear and flake8-quotes. All of these tools can be replaced by ruff, which not only has even more tools included but also executes linting and formatting much faster.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  - repo: https://github.com/charliermarsh/ruff-pre-commit
+    rev: v0.1.2
+    hooks:
+      - id: ruff
+        args: [ --fix, --unsafe-fixes, --exit-non-zero-on-fix ]
+
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.1.2
+    hooks:
+      - id: ruff-format
```
Comment (bonastreyair) [inline .pre-commit-config.yaml:20]: updated to newer version + single pre-commit repo
```suggestion
  - repo: https://github.com/astral-sh/ruff-pre-commit
    rev: v0.1.8
    hooks:
      - id: ruff-format
      - id: ruff
        args: [--fix, --exit-non-zero-on-fix]
```

File: dramatiq/actor.py [hunk 2]
```diff
@@ -54,15 +54,16 @@ class Actor(Generic[P, R]):
       options(dict): Arbitrary options that are passed to the broker
         and middleware.
     """
+
     def __init__(
         self,
-        fn: Callable[P, Union[R, Awaitable[R]]],
+        fn: Callable[P, R | Awaitable[R]],
```
Comment (bonastreyair) [inline dramatiq/actor.py:60]: be aware this things will make older python version to be incompatible too...

Comment (santigandolfo) [reply dramatiq/actor.py:60]: @bonastreyair what do you think?

**Comment by Bogdanp** [comment]
Thanks, but this is too big of a change, and the current tools are fine.

**Pull request closed without merging**